modules/xphoto/src/inpainting_fsr.impl.hpp

```diff
@@ -748,26 +748,28 @@ icvDetermineProcessingOrder(
 
 
 static
-void inpaint_fsr(const Mat &src, const Mat &mask, Mat &dst, const int algorithmType)
+void inpaint_fsr(const Mat &src_, const Mat &mask, Mat &dst, const int algorithmType)
```

Comment:
I propose to pass the matrix by value. cv::Mat is smart pointer, so no large overhead. It solves all potential issues in the pipeline.